### ISSUE TITLE
[Bug] HotReload not working for inner view 

### ISSUE THREAD

**Issue by sheiksyedm** [dotnet/maui#3727]
### Description

We have created a simple custom control that has multiple inner views. HotReload only works for the main custom control view and does not work for inner views. 

Find the code for simple custom control implementation. 

```
public class CusControl : View, IContentView
    {

        private readonly Grid contentGrid;

        public CusControl()
        {
            contentGrid = new Grid();
        }

        public static readonly BindableProperty ChildProperty =
           BindableProperty.Create(nameof(Child), typeof(View), typeof(CusControl), null, propertyChanged: OnChildPropertyChanged);

        public View Child
        {
            get { return (View)this.GetValue(ChildProperty); }
            set { this.SetValue(ChildProperty, value); }
        }

        private static void OnChildPropertyChanged(BindableObject bindable, object oldValue, object newValue)
        {
            var control = bindable as CusControl;
            if (newValue != null)
            {
                control.contentGrid.Children.Add(newValue as View);
            }

            if (oldValue != null)
            {
                control.contentGrid.Children.Remove(oldValue as View);
            }
        }

        object IContentView.Content => contentGrid;

        IView IContentView.PresentedContent => contentGrid;

        Thickness IPadding.Padding => new Thickness(0);

        Size IContentView.CrossPlatformArrange(Rectangle bounds)
        {
            this.ArrangeContent(bounds);
            return bounds.Size;
        }

        Size IContentView.CrossPlatformMeasure(double widthConstraint, double heightConstraint)
        {
            return this.MeasureContent(widthConstraint, heightConstraint);
        }

        // Tried to pass binding context to the content grid. 
        protected override void OnBindingContextChanged()
        {
            base.OnBindingContextChanged();
            if (contentGrid != null)
            {
                SetInheritedBindingContext(this.contentGrid, BindingContext);
            }
          
           
        }
    }
```

### Steps to Reproduce

1. Run the attached sample
2. Try to change the child text values or color, font size, etc.,  that changes won't reflect in the UI (**HotReload issue**). 
3. Now, try to change the CusControl background, HeighRequest, WidthRequest, etc., that changes will reflect in the UI. 

Sample : [HotReload_Check.zip](https://github.com/dotnet/maui/files/7692579/HotReload_Check.zip)

### Version with bug

Preview 10 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows

### Affected platform versions

Android 9 and above, Windows SDK 10.0.19041

**Comment by drasticactions** [comment]
@sheiksyedm If you break the default "LogicalChildren" listing that the default MAUI Elements have, Hot Reload can't work because there's no way for it to know where the children are located.  However, if you implement `IVisualTreeElement` and handle the nested views then it should work, since then it will percolate up the tree.

Basically, if you have Hot Reload enabled and see the Live Visual Tree in Visual Studio, and don't see your elements, Hot Reload will not work. If you add IVisualTreeElement, those items should appear, and then it should start working, so long as you properly handle gets and sets for the control items.

I did an example of this for a custom Window Overlay I created.

https://github.com/drasticactions/DrasticOverlay/blob/main/DrasticOverlay/Overlays/Page/PageOverlay.cs#L9
